datajoint/base_relation.py

```diff
@@ -155,7 +155,14 @@ def insert(self, rows, replace=False, ignore_errors=False, skip_duplicates=False
                 table=self.full_table_name,
                 fields='`'+'`,`'.join(rows.heading.names)+'`',
                 select=rows.make_sql())
-            self.connection.query(query)
+            try:
+                self.connection.query(query)
+            except pymysql.err.InternalError as err:
+                if err.args[0] == server_error_codes['unknown column']:
+                    # args[1] -> Unknown column 'extra' in 'field list'
+                    raise DataJointError('%s : set ignore_extra_fields?' % err.args[1])
```

Comment:
I would expand the suggestion into a full sentence, e.g. "To ignore extra fields, you may set the ignore_extra_fields argument in insert."